questions/126/explanation.md

```diff
@@ -1,3 +1,27 @@
-§[basic.lookup.unqual]¶13 states "A name used in the definition of a static data member of class `X` (...) is looked up as if the name was used in a member function of `X`."
+Will the global `foo` or the member `foo` be chosen?
 
-Even though the call `foo()` occurs outside the class, since `foo` is used in the definition of the static data member `foobar::x`, it is looked up as if `foo()` was called in a member function of `foobar`. If `foo()` was called in a member function of `foobar`, `foobar::foo()` would be called, not the global `foo()`.
\ No newline at end of file
+First, notice that the name `foo` is unqualified. §[basic.lookup.unqual]¶4:
+
+> An *unqualified name* is a name that does not immediately follow a *nested-name-specifier* or the `.` or `->` in a class member access expression (§[expr.ref]), possibly after a `template` keyword or `~`. Unless otherwise specified, such a name undergoes unqualified name lookup from the point where it appears.
+
+What does unqualified name lookup mean? §[basic.lookup.unqual]¶3:
+
+> *Unqualified name lookup* from a program point performs an unqualified search in its immediate scope.
+
+The definition of an unqualified search is given in §[basic.lookup.unqual]¶2:
+
+> An *unqualified search* in a scope `S` from a program point `P` includes the results of searches from `P` in
+>
+> — `S`, and
+>
+> — for any scope `U` that contains `P` and is or is contained by `S`, each namespace contained by `S` that is nominated by a *using-directive* that is active in `U` at `P`.
+>
+> If no declarations are found, the results of the unqualified search are the results of an unqualified search in the parent scope of `S`, if any, from `P`.
+
+So, since we don't have any using directives, we search for `foo` in its immediate scope and if we find any result, we stop searching.
+
+An immediate scope at some program point is the smallest scope that contains it (§[basic.scope.scope]¶1). A declaration's target scope is the immediate scope of its locus (§[basic.scope.scope]¶2). What is the immediate scope at the program point where we call `foo`? §[basic.scope.class]¶1:
+
+> Any declaration of a class or class template `C` introduces a *class scope* that includes the *member-specification* of the *class-specifier* for `C` (if any). For each non-friend redeclaration or specialization whose target scope is or is contained by the scope, the portion after the *declarator-id*, *class-head-name*, or *enum-head-name* is also included in the scope.
+
+So `foo`, which appears after `foobar::x`, whose target scope is the `C`'s class scope, is also included in that scope. We find `foobar::foo` there and don't search any further.
```

Comment:
Can we expand this a little bit (in non-standardese prose?) In "So `foo`, which"  it's not immediately clear that we're talking about the mention of `foo` in `int foobar::x = foo();` It's also a bit unclear how the standard quote above maps to this prose.

Maybe it would help if we start with something like "So in the statement `int foobar::x = foo();`" and then explain how `C`s class scope is the target scope of `foobar::x`, and then how that scope is included when looking up `foo`. I'm not great at scopes and lookup, so I don't have a more concrete suggestion here (but that non-greatness might be useful for review since most other readers might also not be great at this).